lib/ansible/modules/cloud/amazon/iam_cert.py

```diff
@@ -269,10 +280,23 @@ def main():
     cert_chain = module.params.get('cert_chain')
     dup_ok = module.params.get('dup_ok')
     if state == 'present':
-        cert = open(module.params.get('cert'), 'r').read().rstrip()
-        key = open(module.params.get('key'), 'r').read().rstrip()
+        if cert:
+            if os.path.isfile(module.params.get('cert')):
+                cert = open(module.params.get('cert'), 'r').read().rstrip()
+            else:
+                cert = unicode(module.params.get('cert').rstrip())
+
+        if key:
+            if os.path.isfile(module.params.get('key')):
+                key = open(module.params.get('key'), 'r').read().rstrip()
+            else:
+                key = unicode(module.params.get('key').rstrip())
```

Comment:
unicode won't always work (Python 2/3 split) so please use the to_text from ansible's utils instead.